PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/ViewNode.java

```diff
@@ -351,15 +350,87 @@ public void highlight(Pane treePane) {
   }
 
   /**
-   * Remove the highlight of the area of this node.
+   * Remove the highlightNode of the area of this node.
    *
-   * @param treePane the pane from which to remove the highlight area.
+   * @param treePane the pane from which to remove the highlightNode area.
    */
   public void removeHighlight(Pane treePane) {
     treePane.getChildren().remove(highlightArea);
     highlightArea = null;
   }
 
+  /**
+   * Highlight the given node and its child nodes. First search for matches in any of the parent
+   * nodes, then search for matches in the child nodes.
+   *
+   * @param node  the node.
+   * @param color the color.
+   */
+  public void highlightNode(IPhylogeneticTreeNode node, Color color) {
+    IPhylogeneticTreeNode parent = dataNode;
+    while (parent.hasParent()) {
+      parent = parent.getParent();
+      if (parent.equals(node)) {
+        setRecursiveColor(color);
+        return;
+      }
+    }
+    highlightChildNode(node, color);
+  }
+
+  /**
+   * Highlight the given node and its child nodes. Only search for matches in the child nodes.
+   *
+   * @param node  the node.
+   * @param color the color.
+   */
+  private void highlightChildNode(IPhylogeneticTreeNode node, Color color) {
+    if (dataNode.equals(node)) {
+      setRecursiveColor(color);
+    } else {
+      for (ViewNode viewNode : children) {
+        viewNode.highlightNode(node, color);
+      }
+    }
+  }
+
+  /**
+   * Recursively set the color of this node and all child nodes to the given color.
+   *
+   * @param color the color.
+   */
+  private void setRecursiveColor(Color color) {
+    if (dataNode.getDirectChildCount() == 0) {
+      setFill(color.darker().darker());
+    } else {
+      setFill(color);
+      for (ViewNode viewNode : children) {
+        viewNode.setRecursiveColor(color);
+      }
+    }
+  }
+
+  /**
+   * Recursively set the color of this node and all child nodes to the default color.
+   */
+  public void removeHighlightColor() {
+    setDefaultColor();
+    for (ViewNode viewNode : children) {
+      viewNode.removeHighlightColor();
+    }
+  }
+
+  /**
+   * Set the default color of this node.
+   */
+  private void setDefaultColor() {
+    if (dataNode.getDirectChildCount() == 0) {
```

Comment:
This is sort of like doing `size() == 0` vs `isEmpty()` on a `List<E>`.

Since you treat this case like a leaf node (you give it `LEAF_COLOR`), maybe make a method like `isLeaf()` which returns this condition?